nvflare/fuel/f3/streaming/byte_streamer.py

```diff
@@ -71,200 +82,233 @@ def __init__(
         self.offset_ack = 0
         self.secure = secure
         self.optional = optional
+        self.stopped = False
 
-    def __str__(self):
-        return f"Tx[SID:{self.sid} to {self.target} for {self.channel}/{self.topic}]"
+        self.stream_future = StreamFuture(self.sid)
+        self.stream_future.set_size(stream.get_size())
 
+        self.window_size = CommConfigurator().get_streaming_window_size(STREAM_WINDOW_SIZE)
```

Comment:
so stream size is the total size of the payload to be sent?
what about "stream_window_size" and "chunk_size"?

Reply:
chunk_size is how big to send for each chunk (we break the payload to chunks)
and stream_window_size is how many chunks can be sent/handled at the same time?